bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -128,6 +130,123 @@ enum ClusterConfig {
         region: String,
     },
 }
+enum IPFamily {
+    IPv4,
+    IPv6,
+}
+
+fn get_family_string(family: IPFamily) -> &'static str {
+    match family {
+        IPFamily::IPv4 => "IPv4",
+        IPFamily::IPv6 => "IPv6",
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Config {
+    api_version: String,
+    kind: String,
+    metadata: Metadata,
+    availability_zones: Vec<String>,
+    kubernetes_network_config: KubernetesNetworkConfig,
+    addons: Vec<Addon>,
+    iam: IAMConfig,
+    managed_node_groups: Vec<ManagedNodeGroup>,
+}
+
+#[derive(Serialize)]
+struct Metadata {
+    name: String,
+    region: String,
+    version: String,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct KubernetesNetworkConfig {
+    ip_family: String,
```

Comment:
As long as your enum is given `#[derive(Serialize)]` serde should treat it properly (even if you don't implement `to_string()` as above. If we don't set this to enum, we don't get the benefits of type checking that the enum is trying to grant us.